### ISSUE TITLE
Module shared_vpc_access causes v12.0 causes unnecessary changes

### ISSUE THREAD

**Issue by cpf-hse** [terraform-google-modules/terraform-google-project-factory#698]
### TL;DR

When switching from Version 11.3.1 to 12.0 terraform plan delivered 34 to add and 34 to destroy without any other changes.
Going back to version 11.3.1 terraform plan shows "No Changes"

All changes looked like this 
(Please not I renamed real project names, project ids, subnets etc.)

 # module.shared_vpc.module.shared_vpc_access["PROJECT1"].google_compute_subnetwork_iam_member.service_shared_vpc_subnet_users[0] must be replaced
-/+ resource "google_compute_subnetwork_iam_member" "service_shared_vpc_subnet_users" {
      ~ etag       = "BwW9kXb6SBI=" -> (known after apply)
      ~ id         = "projects/our-shared-network/regions/europe-west3/subnetworks/subnet-1/roles/compute.networkUser/serviceaccount:service-PROJECTID@container-engine-robot.iam.gserviceaccount.com" -> (known after apply)
      ~ subnetwork = "projects/our-shared-network/regions/europe-west3/subnetworks/subnet1" -> "subnet1" # forces replacement
        # (4 unchanged attributes hidden)
    }

### Expected behavior

No changes. Your infrastructure matches the configuration.

### Observed behavior

Plan: 34 to add, 0 to change, 34 to destroy.

### Terraform Configuration

```hcl
module "shared_vpc_access" {
  for_each = var.service_projects
  source   = "terraform-google-modules/project-factory/google//modules/shared_vpc_access"

  enable_shared_vpc_service_project = true
  host_project_id                   = var.host_project
  service_project_id                = each.key
  active_apis = [
    "compute.googleapis.com",
    "container.googleapis.com"
  ]
  shared_vpc_subnets = [
    "projects/our-shared-network/regions/europe-west3/subnetworks/subnet-1",
    "projects/our-shared-network/regions/europe-west3/subnetworks/subnet-2",
  ]

  depends_on = [
    google_project_service.host_vpc_container,
    google_project_service.example-host_vpc_servicenetworking
  ]
}
```

### Terraform Version

```sh
terraform {
  required_version = ">= 0.14"
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "~> 3.50"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = "~> 3.54"
    }
```

**Comment by bharathkkb** [comment]
@cpf-hse Were you also upgrading provider versions to `~> 4.5` along side the project factory upgrade?

**Comment by cpf-hse** [comment]
@bharathkkb I came across this situation while migrating a terraform project to GitLab with CI/CD pipeline. There was no intention on upgrading anything. I was also able to reproduce this error on a "fresh" git repo where I had to run terraform init first. My "old" local project still used version 11.3.1.

The problem was caused by not specifying a version for the terraform-google-modules/project-factory/google//modules/shared_vpc_access. It just used the latest version that caused the unexpected changes.

**Comment by cpf-hse** [comment]
When trying to upgrade to the latest provider version this error still occurs.
